proxy/proxy.go

```diff
@@ -55,6 +71,22 @@ func MiddlewareSetPrefixHeader() Middleware {
 	}
 }
 
+func MiddlewareSetTempoPrefixHeader() Middleware {
+	return func(r *http.Request) {
+		prefix, ok := getPrefix(r.Context())
+		if !ok {
+			return
+		}
+
+		// Do not override the prefix header if it is already set.
+		if r.Header.Get(PrefixHeader) != "" {
+			return
+		}
+
+		r.Header.Set(PrefixHeader, prefix)
```

Comment:
Should this be `TempoPrefixHeader`? I can't find any usage of the new `TempoPrefixHeader` constant.